src/DevCenter/DevCenterData.AutoRest/README.md

```diff
@@ -0,0 +1,147 @@
+title: DevCenterdata
+subject-prefix: DevCenter
+endpoint-resource-id-key-name: https://devcenter.azure.com
+directive:
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/devboxes/{devBoxName}"].delete.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/devboxes/{devBoxName}:start"].post.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/devboxes/{devBoxName}:stop"].post.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/devboxes/{devBoxName}:restart"].post.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/environments/{environmentName}"].put.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "#/definitions/Environment"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/environments/{environmentName}"].delete.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/devboxes"].get.operationId
+    transform: return "DevBoxes_ListAllDevBoxes"
+  - from: swagger-document
+    where: $.paths["/users/{userId}/devboxes"].get.operationId
+    transform: return "DevBoxes_ListAllDevBoxesByUser"
+  - where:
+      subject: ^(.*)(DevBoxPool)(.*)$
+    set:
+      subject: Pool
+  - where:
+      subject: ^(.*)(DevBoxSchedule)(.*)$
+    set:
+      subject: Schedule
+  - where:
+      subject: ^(.*)(DevCenterProject)(.*)$
+    set:
+      subject: Project
+  - where:
+      subject: ^(.*)(EnvironmentCatalog)(.*)$
+    set:
+      subject: Catalog
+# Matches any subject that is not strictly "DevBox" or "Environment" (eg. still includes DevBoxAction)
+  - where:
+      subject: ^(?!DevBox$|Environment$).*
+      parameter-name: UserId
+    set:
+      default:
+        script: '"me"'
+# Matches cmdlets with exact subject DevBox or Environment, but not with verb Get
+  - where:
+      verb: ^(?!Get$)
```

Comment:
could you explain which will be matched?

Reply:
This will match all cmdlets that are strictly "DevCenter" or "Environment" but not with the verb "Get". This is because we want set the default parameter of "UserId" to "me" in all these cases except for the "Get" cmdlets. 
It will correspond to the following cmdlets: 
Deploy-AzDevCenterAdminEnvironment
New-AzDevCenterAdminDevCenter
New-AzDevCenterUserEnvironment
Remove-AzDevCenterAdminDevCenter
Remove-AzDevCenterUserEnvironment